### PR TITLE
feat(Testing): Added test cases for our dataset.

### PR DESCRIPTION
**Pull request by szemyd**

### CODE REVIEW

File: run_preprocessing.py [hunk 1]
```diff
@@ -289,6 +289,10 @@ def extract_reverse_edges(transactions: pd.DataFrame) -> dict:
     return transactions.groupby("article_id")["customer_id"].apply(list).to_dict()
 
 
+def extract_edges_articles(transactions: pd.DataFrame) -> dict:
```
Comment (itchingpixels) [inline run_preprocessing.py:292]: we don't need this!

File: data/data_loader.py [hunk 2]
```diff
@@ -23,12 +23,14 @@ def create_dataloaders(
         config=config,
         edge_path=data_dir + "edges_train.pt",
         graph_path=data_dir + "train_graph.pt",
+        article_edge_path=data_dir + "rev_edges_train.pt",
```
Comment (itchingpixels) [inline data/data_loader.py:26]: not needed

File: data/dataset.py [hunk 3]
```diff
@@ -71,7 +196,11 @@ def __getitem__(self, idx: int) -> Union[Data, HeteroData]:
         user_features = self.graph[Constants.node_user].x[idx]
 
         # Prepare connected article features
-        article_features = self.graph[Constants.node_item].x[all_touched_edges]
+        article_features = self.graph[Constants.node_item].x[
+            torch.sort(all_touched_edges)[0]
+        ]
+        print("FASDFSDFAD")
```
Comment (itchingpixels) [inline data/dataset.py:202]: probably not needed

File: data/dataset.py [hunk 4]
```diff
@@ -90,48 +219,101 @@ def __getitem__(self, idx: int) -> Union[Data, HeteroData]:
...
+        sampled_edges_negative[0] = id_tensor.repeat(len(sampled_edges_negative[0]))
+
+        # Remap negative edges to start from zero
+        all_touched_edges_ids = torch.unique(all_touched_edges[1])
+
+        sampled_edges_negative[1] = remap_indexes_to_zero(
+            sampled_edges_negative[1], buckets=all_touched_edges_ids
+        )
+        # Shift negative edges to be after the subgraph edges
+        # sampled_edges_negative[1] = torch.add(
```
Comment (itchingpixels) [inline data/dataset.py:312]: is this needed?

File: data/data_loader.py (hunk 2 above)
Comment (itchingpixels) [reply data/data_loader.py:26]: sorry, my bad!

File: data/dataset.py (hunk 4 above)
Comment (szemyd) [reply data/dataset.py:312]: Not needed!

File: data/data_loader.py (hunk 2 above)
Comment (szemyd) [reply data/data_loader.py:26]: This is needed for my dataset.

File: data/dataset.py (hunk 3 above)
Comment (szemyd) [reply data/dataset.py:202]: yes
Also took out a bunch of superflous remapping in first_user

**Pull request merged**